src/data/real/cau_seq.lean

```diff
@@ -198,13 +198,24 @@ instance : has_zero (cau_seq β abv) := ⟨const 0⟩
 instance : has_one (cau_seq β abv) := ⟨const 1⟩
 instance : inhabited (cau_seq β abv) := ⟨0⟩
 
-@[simp] theorem zero_apply (i) : (0 : cau_seq β abv) i = 0 := rfl
-@[simp] theorem one_apply (i) : (1 : cau_seq β abv) i = 1 := rfl
+@[simp, norm_cast] theorem zero_apply (i) : (0 : cau_seq β abv) i = 0 := rfl
+@[simp, norm_cast] theorem one_apply (i) : (1 : cau_seq β abv) i = 1 := rfl
 @[simp] theorem const_zero : const 0 = 0 := rfl
 
 theorem const_add (x y : β) : const (x + y) = const x + const y :=
 rfl
 
+private lemma nsmul_aux (n : ℕ) (f : cau_seq β abv) (i : ℕ) : nsmul_rec n f i = n • f i :=
+by induction n; simp only [*, nsmul_rec, zero_apply, add_apply, zero_nsmul, succ_nsmul]
+
+instance has_nsmul : has_smul ℕ (cau_seq β abv) :=
```

Comment:
Does this generalize to any action that acts on ` ℚ`? Then you wouldn't need to prove zsmul separately, and could add an `is_cau_seq.smul` lemma. If not, then a `is_cau_seq.nsmul` lemma would still be nice.

Reply:
`ℚ` doesn't act on `β` in general (it's only a ring), so I'm not sure how that would make sense. I think it's true when it factors through `ℤ`, but we don't have access to `ordered_smul` here so I'm not sure how that could work.

Reply:
Ah you're right, `has_ordered_smul` is probably the appropriate generalization (and ` α` rather than ` β `. Are there complications with importing it?

Reply:
It's quite a bigger import, but I can still add it.

Reply:
Is it that big? It feels like a smaller import than whatever brings in `linear_ordered_field` which this file already uses

Reply:
It imports `algebra.order.field`, `data.set.pointwise`, `tactic.positivity`

Reply:
We already import the first one, I think arguably the second one should have it's import direction flipped, and the third one is probably ok

Reply:
Actually, I'm not sure that's worth doing because that would mean either:
* doing a custom proof
* porting the existing proof to `is_cau_seq` (which seems to have very few uses), including `has_neg`, `has_mul`, `bounded'`.

It seems like we should just delete `is_cau_seq` in favor of making `cau_seq` a custom structure.

Reply:
Let's not make that deletion in this PR, but I somewhat see your point. Are you also opposed to the `ordered_smul` generalization?

Reply:
No, but it's not obvious to me how it would go. I need to think about it.

Reply:
This does the job:
```
instance has_smul [has_smul α β] [is_scalar_tower α β β] :
  has_smul α (cau_seq β abv) :=
⟨λ n f, of_eq (const (n • 1) * f) (n • f) (λ i, smul_one_mul n (f i))⟩
```
Presumably there are better generalizations, but that's enough to cover actions by nat, int, and rat.

Reply:
Very nice! and much weaker than expected.